File: 프로그래머스/Level1/소수찾기.cpp
#include <string>
#include <vector>

using namespace std;
int arr[1000000];

int solution(int n) {
    int answer = 0;
    
    for(int i=2; i<=n; i++){
        arr[i]=i;
    }
    for(int i=2; i<=n; i++){
        if(arr[i]==0) continue;
        for(int j=i+i; j<=n; j+=i){
            arr[j]=0;
        }
    }
    for(int i=2; i<=n; i++){
        if(arr[i]!=0){
            answer ++;
        }
    }
    return answer;
}

File: 프로그래머스/Level1/소수 찾기.js
function solution(n) {
    let answer = 0;
    
    function primeNum(number){
        if(number === 2) return true;
        for(let i=2; i<=Math.floor(Math.sqrt(number)); i++){
            if(number%i === 0) return false;
        }
        return true;
    }

    for(let j=2; j<=n; j++)
        if(primeNum(j)) answer ++;
    
    return answer;
}
